### ISSUE TITLE
math preview: option to use preamble & environments of the current .tex file

### ISSUE THREAD

**Issue by moiraine-sedai** [SublimeText/LaTeXTools#977]
Hi, thanks a lot for the new math/figure preview feature! However, as has been reported in [#915](https://github.com/SublimeText/LaTeXTools/issues/915), if my current latex document has some custom preamble or environment, e.g.
`\newcommand{\Prob}{\mathbf{P}}`
`\usepackage{braket}`
then the preview for equations containing these symbols are not generated properly. In [#915](https://github.com/SublimeText/LaTeXTools/issues/915), this is classified as 'performance issue' and reference to [#896](https://github.com/SublimeText/LaTeXTools/issues/896). However, the solution in that issue is not very good: one has to provide a custom preamble, almost hard-coded in the preference file or link to another file. How about just use the preamble and environments defined in the current .tex file? Thanks a lot!

**Comment by r-stein** [comment]
Yes, that's somewhere on the TODO list with a lower priority. It would be easy to grap the usepackage's and newcommand's and put them in the preview document, but that would significantly slow down Sublime Text. It can also be done without slowing ST, but will require more work.

In addition I am still thinking about what to put inside, especially with the used packages. Most packages slows down the compile time, but don't affect the math preview at all. So we may select the packages a little bit smarter. In addition the user can just provide a superset of all math packages he usually uses.
However copying the `DeclareMathOperator` and `newcommand` commands would be nice.

I am open to input =)

**Comment by randy3k** [comment]
Before it is properly addressed in LaTeXTools, you could install [LaTeX][1] and use `LaTeXZeta: Toggle Auto Set Preview Math Template Preamble` the option to automatically include 
`DeclareMathOperator` and `newcommand` of the current file.

[1]: https://github.com/randy3k/LaTeXZeta#others

**Comment by ViktorQvarfordt** [comment]
Why not just copy the users entire preamble? Inspired by LaTeXZeta I'm currently using

```python
import sublime, sublime_plugin
import re

class LaTeXToolsPreviewMathUseEntirePreamble(sublime_plugin.EventListener):

    def on_activated_async(self, view):
        self.set_template_preamble(view)

    def on_post_save_async(self, view):
        self.set_template_preamble(view)

    def set_template_preamble(self, view):
        # Make sure we are in a LaTeX file.
        if not view.match_selector(0, "text.tex.latex"):
            return
        # Extract entire preamble
        view_text = view.substr(sublime.Region(0, view.size()))
        preamble = re.search(r'\\documentclass(?:[^}]*)}(.*?)\\begin{document}', view_text, re.DOTALL).group(1)
        # Update settings
        view.settings().set("preview_math_template_preamble", preamble)
```
